hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -293,6 +361,38 @@ protected void commit(HoodieTable table, String commitActionType, String instant
         });
   }
 
+  /**
+   * Write to metadata table.
+   * We could either have streaming writes enabled or disabled.
+   * Incase of streaming writes, we have to write to FILES partition and wrap up the commit in metadata table.
+   * If not for streaming writes, we take the legacy route, where we start a new delta commit in metadata table, prepare all records and write to it and wrap up the commit.
+   * @param skipStreamingWritesToMetadataTable true if streaming writes to metadata needs to be skipped. Eg, row writer flow does not support streaming writes to metadata table.
+   * @param config instance of {@link HoodieWriteConfig}.
+   * @param table instance of {@link HoodieTable}.
+   * @param writeOperationType write operation type.
+   * @param instantTime instance time of interest.
+   * @param metadataWriteStatsSoFar List of HoodieWriteStats referring to completed writes to metadata table so far. If streaming writes are disabled, this will be empty.
+   * @param metadata instance of {@link HoodieCommitMetadata} of interest.
+   */
+  private void writeToMetadataTable(boolean skipStreamingWritesToMetadataTable, HoodieWriteConfig config, HoodieTable table, WriteOperationType writeOperationType,
```

Comment:
if streaming writes are enabled, this will write to FILES partition and commit in metadata table. 
if not, we take the legacy way of preparing records to all metadata table partitions and committing to mdt.